### PR TITLE
Make the linter happy

### PR DESCRIPTION
**Pull request by badboy**
ktlint was accumulating some non-allowed stuff which we can easily get rid off.
(we allow up to 10 lint failures apparently)

### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `make test` runs without emitting any warnings
  - `make lint` runs without emitting any errors
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry or an explanation of why it does not need one
  - Any breaking changes to language binding APIs are noted explicitly

### CODE REVIEW

File: glean-core/android/src/main/java/mozilla/telemetry/glean/private/DatetimeMetricType.kt [hunk 1]
```diff
@@ -102,8 +102,10 @@ class DatetimeMetricType internal constructor(
                 hour = value.get(Calendar.HOUR_OF_DAY),
                 minute = value.get(Calendar.MINUTE),
                 second = value.get(Calendar.SECOND),
-                nano = value.get(Calendar.MILLISECOND) * 1000000L,
-                offset_seconds = value.get(Calendar.ZONE_OFFSET).div(1000)
+                nano = java.util.concurrent.TimeUnit.MILLISECONDS.toNanos(value.get(Calendar.MILLISECOND).toLong()),
```
Comment (Dexterp37) [inline glean-core/android/src/main/java/mozilla/telemetry/glean/private/DatetimeMetricType.kt]: nit: you could do `import java.util.concurrent.TimeUnit as AndroidTimeUnit` above in the file and then use `AndroidTimeUnit.MILLISECONDS.toNanos` to make this a bit more readable

Comment (badboy) [reply glean-core/android/src/main/java/mozilla/telemetry/glean/private/DatetimeMetricType.kt]: Ha, that was the missing piece.

File: glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt [hunk 2]
```diff
@@ -85,7 +85,10 @@ open class GleanInternalAPI internal constructor () {
         this.configuration = configuration
 
         this.gleanDataDir = File(applicationContext.applicationInfo.dataDir, GLEAN_DATA_DIR)
-        handle = LibGleanFFI.INSTANCE.glean_initialize(this.gleanDataDir.path, applicationContext.packageName, uploadEnabled.toByte())
+        handle = LibGleanFFI.INSTANCE.glean_initialize(
+            this.gleanDataDir.path,
+            applicationContext.packageName,
+            uploadEnabled.toByte())
```
Comment (mdboom) [inline glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt]: Maybe final `)` on the next line?  Maybe ktlint doesn't care, but I think that's more readable (and consistent with rustfmt FWIW).

**Comment by badboy** [comment]
Ah, detekt it is then. There are still 2 failures left (too many return statements).
I will probably need to add an exception marker there, but I will take that in a follow-up

**Comment by badboy** [comment]
I ended up adding it here as well

**Pull request merged**